sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ReadFromKafkaDoFn.java

```diff
@@ -518,7 +525,7 @@ private ConsumerRecords<byte[], byte[]> poll(
         return rawRecords;
       }
       elapsed = sw.elapsed();
-      if (elapsed.toMillis() >= KAFKA_POLL_TIMEOUT.toMillis()) {
+      if (elapsed.toMillis() >= consumerPollingTimeout.toMillis()) {
         // timeout is over
         return rawRecords;
```

Comment:
In the case where the `rawRecords.isEmpty()`, might be worth adding some sort of count that tracks how many times `rawRecords` has been empty. If this count is too high, maybe emit a warning log that it is happening too often and that perhaps they should increase the poll timeout

Reply:
added warning when no messages polled.